### PR TITLE
QOL updates from running benchmark samples

### PR DESCRIPTION
**Pull request by granawkins**
Changes that I made while working with samples/benchmarks.

### CODE REVIEW

File: benchmarks/arg_parser.py [hunk 1]
```diff
@@ -24,6 +24,13 @@ def common_benchmark_parser():
             " depends on benchmark."
         ),
     )
+    parser.add_argument(
+        "--directory",
```
Comment (PCSwingle) [inline benchmarks/arg_parser.py:28]: Could we makes this --sample-directory? I think directory is too vague for this.

File: benchmarks/benchmark_result.py [hunk 2]
```diff
@@ -83,5 +86,5 @@ def to_json(self):
         return json.dumps(self.to_dict())
 
     @classmethod
-    def from_json(cls, json_str):
+    def from_json(cls, json_str: str) -> "BenchmarkResult":
```
Comment (PCSwingle) [inline benchmarks/benchmark_result.py:89]:
```suggestion
    def from_json(cls, json_str: str) -> Self:
```

Comment (jakethekoenig) [reply benchmarks/benchmark_result.py]: Looks like this isn't a feature in python 3.11?

**Review by jakethekoenig — APPROVED** [review]

File: benchmarks/benchmark_result.py (hunk 2 above)
Comment (granawkins) [reply benchmarks/benchmark_result.py]: ya fwiw this broke the macos tests, I switch to `from __future__`..

**Pull request merged**